### PR TITLE
Add Microsoft.Botkit project

### PR DESCRIPTION
**Pull request by GasparAcevedoZainSouthworks**
**Incoming changes**

* Removed old _Botkit_ Project (_BotkitProject.csproj_)
* Added _Microsoft.BotKit_ project, which is **_.NetStandard 2.0_**
* Add MIT license copyright to _Botkit_'s files
* Fixed some _TS_ to _C#_ migrations in the _Botkit_ Project

**Note**
There are still some porting labeled as _TO-DO_ in the Botkit project, which should be solved in next PRs/

### CODE REVIEW

File: C#/Botkit/Microsoft.BotKit/BotWorker.cs [hunk 1]
```diff
@@ -22,7 +21,7 @@ public class BotWorker
         /// </summary>
         /// <param name="controller">A pointer to the main Botkit controller</param>
         /// <param name="config">An object typically containing { dialogContext, reference, context, activity }</param>
-        public BotWorker(Botkit controller, object config)
+        public BotWorker(Core.Botkit controller, object config)
```
Comment (VictorGrycuk) [inline C#/Botkit/Microsoft.BotKit/BotWorker.cs]: `Core` in `Core.Botkit` can be removed
image

File: C#/Botkit/Microsoft.BotKit/CMS/BotkitCMSHelper.cs [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+using Microsoft.BotKit.Core;
 using Microsoft.Bot.Builder.Dialogs;
 
-namespace BotkitLibrary.CMS
+namespace Microsoft.BotKit.CMS
 {
     public class BotkitCMSHelper
     {
-        public BotkitCMSHelper(Botkit botkit, CMSConfiguration configuration)
+        public BotkitCMSHelper(Core.Botkit botkit, CMSConfiguration configuration)
```
Comment (VictorGrycuk) [inline C#/Botkit/Microsoft.BotKit/CMS/BotkitCMSHelper.cs]: Same comment about removing `Core` here

Comment (GasparAcevedoZainSouthworks) [reply C#/Botkit/Microsoft.BotKit/CMS/BotkitCMSHelper.cs]: Feedback Applied

File: C#/Botkit/Microsoft.BotKit/BotWorker.cs (hunk 1 above)
Comment (GasparAcevedoZainSouthworks) [reply C#/Botkit/Microsoft.BotKit/BotWorker.cs]: Feedback applied

**Comment by GasparAcevedoZainSouthworks** [comment]
> Small change request

Feedback Applied

File: C#/Botkit/Botkit.sln [hunk 3]
```diff
@@ -3,24 +3,24 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 16
 VisualStudioVersion = 16.0.28803.352
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "BotkitLibrary", "Botkit\BotkitLibrary.csproj", "{27A6C425-DA23-4D95-B754-9A4BA6C2CE46}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "botbuilder-slack-adapter", "botbuilder-slack-adapter\botbuilder-slack-adapter.csproj", "{E2B0D367-41B9-406E-90A8-89C7648A926A}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "botbuilder-slack-adapter", "botbuilder-slack-adapter\botbuilder-slack-adapter.csproj", "{E2B0D367-41B9-406E-90A8-89C7648A926A}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.BotKit", "Microsoft.BotKit\Microsoft.BotKit.csproj", "{9A57B26C-9725-4A25-89FE-760B657E4F61}"
```
[the remark this answers is not in the document]
Comment (GasparAcevedoZainSouthworks) [reply C#/Botkit/Botkit.sln]: This will be applied in another PR

**Pull request merged**